File: app/src/main/java/com/example/proyectoandroidtdp/Conversor/Conversor.java
package com.example.proyectoandroidtdp.Conversor;

@SuppressWarnings("StringConcatenationInLoop")
public class Conversor implements ConversorAbstracto {

    public Conversor(){
    }

    public int[] stringToArray(String msg) {
        int[] array = new int[msg.length()];
        for(int i = 0; i < array.length; i++)
            array[i] = Integer.parseInt(Character.toString(msg.charAt(i)));
        return array;
    }

    public String arrayToString(int[] array) {
        String toReturn = "";
        for(int i : array)
            toReturn += i;
        return toReturn;
    }
}

File: app/src/main/java/com/example/proyectoandroidtdp/conversor/Conversor.kt
package com.example.proyectoandroidtdp.conversor

class Conversor : ConversorAbstracto {
    override fun arrayToString(array: IntArray): String {
        var toReturn = ""
        for (i in array) toReturn += i
        return toReturn
    }

    override fun stringToArray(msg: String): IntArray {
        val array = IntArray(msg.length)
        for (i in array.indices) array[i] = msg[i].toString().toInt()
        return array
    }
}